infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/plugins/DefaultClusterManager.java

```diff
@@ -181,57 +229,161 @@ public TopologyConfigurationMsg queryTopologyConfig(boolean useCached) {
+    public TopologyDescriptor generateConfigWithRoleSwitch() {
+        TopologyDescriptor currentConfig = new TopologyDescriptor(topologyConfig);
+
+        List<ClusterDescriptor> newActiveClusters = new ArrayList<>();
+        List<ClusterDescriptor> newStandbyClusters = new ArrayList<>();
+        currentConfig.getActiveClusters().values().forEach(activeCluster ->
+                newStandbyClusters.add(new ClusterDescriptor(activeCluster, ClusterRole.STANDBY)));
+        for (ClusterDescriptor standbyCluster : currentConfig.getStandbyClusters().values()) {
+            if (newActiveClusters.isEmpty()) {
+                newActiveClusters.add(new ClusterDescriptor(standbyCluster, ClusterRole.ACTIVE));
             } else {
-                standbyClusters.add(standbyCluster);
+                newStandbyClusters.add(new ClusterDescriptor(standbyCluster, ClusterRole.STANDBY));
             }
         }
 
-        TopologyDescriptor newSiteConf = new TopologyDescriptor(1L, Arrays.asList(newPrimary), standbyClusters);
-        return newSiteConf;
+        return new TopologyDescriptor(++configId, newActiveClusters, newStandbyClusters);
+    }
+
+    /**
+     * Create a new topology config, which marks all standby cluster as active on purpose.
+     * System should drop messages between any two active clusters.
+     **/
+    public TopologyDescriptor generateConfigWithAllActive() {
+        TopologyDescriptor currentConfig = new TopologyDescriptor(topologyConfig);
+        ClusterDescriptor currentActive = currentConfig.getActiveClusters().values().iterator().next();
+
+        List<ClusterDescriptor> newActiveClusters = new ArrayList<>();
+        currentConfig.getStandbyClusters().values().forEach(standbyCluster ->
+                newActiveClusters.add(new ClusterDescriptor(standbyCluster, ClusterRole.ACTIVE)));
+        newActiveClusters.add(currentActive);
+
+        return new TopologyDescriptor(++configId, newActiveClusters, new ArrayList<>());
+    }
+
+    /**
+     * Create a new topology config, which marks all cluster as standby on purpose.
+     * System should not send messages in this case.
+     **/
+    public TopologyDescriptor generateConfigWithAllStandby() {
+        TopologyDescriptor currentConfig = new TopologyDescriptor(topologyConfig);
+        ClusterDescriptor currentActive = currentConfig.getActiveClusters().values().iterator().next();
+
+        List<ClusterDescriptor> newStandbyClusters = new ArrayList<>(currentConfig.getStandbyClusters().values());
+        ClusterDescriptor newStandby = new ClusterDescriptor(currentActive, ClusterRole.STANDBY);
+        newStandbyClusters.add(newStandby);
+
+        return new TopologyDescriptor(++configId, new ArrayList<>(), newStandbyClusters);
+    }
+
+    /**
+     * Create a new topology config, which marks all standby cluster as invalid on purpose.
+     * System should not send messages in this case.
+     **/
+    public TopologyDescriptor generateConfigWithInvalid() {
```

Comment:
maybe we should also add a case with active also as invalid

Reply:
Sounds good, will add it in the next PR.